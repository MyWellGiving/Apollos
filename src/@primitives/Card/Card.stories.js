import React from 'react';
import { View } from 'react-native';
import { storiesOf } from '@storybook/react-native';


import Card from './';

storiesOf('@primitives/Card', module)
  .add('Default', () => {
    const centered = {
      justifyContent: 'center',
      alignItems: 'center',
      flex: 1,
    };

    return (
      <View style={centered}>
        <Card />
      </View>
    );
  });
