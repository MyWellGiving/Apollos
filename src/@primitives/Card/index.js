import React from 'react';
import { View } from 'react-native';
import { compose, pure, setPropTypes } from 'recompose';
import PropTypes from 'prop-types';
import styled from '@primitives/styled';

const enhance = compose(
  pure,
  setPropTypes({
    children: PropTypes.node,
    style: View.propTypes.style,
  }),
);

// const StyledCard = styled(({ theme }) => ({
const StyledCard = styled(() => ({
  height: 400,
  width: '92%',
  borderWidth: 1,
  borderRadius: 6,
}))(View);

const Card = enhance(({
  children,
  style: styleProp = {},
  ...otherProps
}) => (
  <StyledCard
    style={styleProp}
    {...otherProps}
  >
    {children}
  </StyledCard>
));

export default Card;
