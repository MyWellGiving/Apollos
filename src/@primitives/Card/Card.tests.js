import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from '@primitives/ThemeProvider';
import Card from './';

describe('the Card component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Card />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
