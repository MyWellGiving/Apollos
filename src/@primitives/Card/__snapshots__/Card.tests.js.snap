// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Card component should render 1`] = `
<View
  style={
    Object {
      "borderRadius": 6,
      "borderWidth": 1,
      "height": 400,
      "width": "92%",
    }
  }
/>
`;
